DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/devlink/test_devlink_sct.py

```diff
@@ -0,0 +1,274 @@
+                      'want_ip': want_ip,
+                      'want_port': want_port,
+                      'want_tcp': choice([True, False]) if want_port else False,
+                      'want_vlan_ethtype': want_ip and want_vlan}
+
+    tc_rules = {}
+    for port_indx in range(ports_num):
+        tc_rule = tcutil_generate_rule_with_random_selectors(dut_ports[port_indx], **rule_selectors)
+        original_rule = deepcopy(tc_rule)
+        randomize_rule_by_src_dst_field(tc_rule, rule_selectors)
+        tc_rules[dut_ports[port_indx]] = {'tc_rule': tc_rule, 'original_rule': original_rule}
+
+    # 4.Prepare traffic matching all SCT supported traps and transmit
+    streams = {}
+    for port_indx in range(ports_num):
+        src_dst_ports = [tg_ports[port_indx], tg_ports[port_indx + 1]] if not port_indx else [tg_ports[port_indx], tg_ports[port_indx - 1]]
+        stream = await get_sct_streams(dev_groups, src_dst_ports, dent_dev, dut_ports[port_indx])
+        streams.update(stream)
+        custome_stream = {f'custome_stream_{dut_ports[port_indx]}': {
+                        'type': 'raw',
+                        'protocol': '802.1Q' if want_vlan else tc_rule['protocol'],
+                        'ip_source': dev_groups[src_dst_ports[0]][0]['name'],
+                        'ip_destination': dev_groups[src_dst_ports[1]][0]['name'],
+                        'srcMac': tc_rules[dut_ports[port_indx]]['original_rule']['filtertype']['src_mac'],
+                        'dstMac': tc_rules[dut_ports[port_indx]]['original_rule']['filtertype']['dst_mac'],
+                        'frameSize': frame_size,
+                        'rate': SCT_MAP['acl_code_3']['exp'] + 2000}
+                         }
+        overwrite_src_dst_stream_fields(custome_stream, tc_rules[dut_ports[port_indx]]['original_rule'], rule_selectors)
+        streams.update(custome_stream)
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(20 if ports_num == 4 else 10)
+
+    # 5.For each stream, verify CPU rate is as expected
+    coroutines_cpu_stat = []
+    coroutines_devlink = []
+    for trap_name, sct in SCT_MAP.items():
+        if trap_name == 'acl_code_3':
+            continue
+        # Due to absence of ability to suspend streams, increase devitation for sct traffic with expected rate around <=200
+        # in case when we send traffic from 4 ports at once
+        deviation = 0.25 if sct['exp'] <= 200 and ports_num == 4 else DEVIATION
+        coroutines_cpu_stat.append(verify_cpu_traps_rate_code_avg(dent_dev, sct['cpu_code'], sct['exp'], deviation=deviation, logs=True))
+        coroutines_devlink.append(verify_devlink_cpu_traps_rate_avg(dent_dev, trap_name, sct['exp'], deviation=deviation, logs=True))
+    await asyncio.gather(*coroutines_cpu_stat)
+    await asyncio.sleep(5)
+    await asyncio.gather(*coroutines_devlink)
```

Comment:
why do you need to separate these coroutines? why can't you use a single `asyncio.gather()`?

Reply:
Because without separation it fails, I tried and asyncio failed to launch 30 cmds in parallel on DUT

Reply:
as discussed, add a comment so it is more obvious